### DISCUSSION TITLE
FastAPI 0.87.0 (and in particular starlette 0.21.0) broke query string parsing in TestClient (starlette 0.22.0 fixes it!)

### DISCUSSION THREAD

**Discussion by andreagrandi** [fastapi/fastapi#6199] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
```

### Description

We have a test that does something like this:
```python
now = utc_datetime()
url = client.app.url_path_for("balances")
start_date = now - timedelta(minutes=1)
start_date_param = urlencode({"start_date": start_date.isoformat()})
end_date_param = urlencode({"end_date": now.isoformat()})
response = client.get(url + f"?{start_date_param}&{end_date_param}")
```
With FastAPI 0.86 everything works. Starting from 0.87 (which uses `starlette` 0.21.0) our endpoint errors with:
```python
{'error': {'code': 'validation_error', 'message': 'Validation Error', 'errors': [{'attribute': 'query -> start_date', 'code': 'value_error.datetime', 'message': 'invalid datetime format', 'ctx': None}, {'attribute': 'query -> end_date', 'code': 'value_error.datetime', 'message': 'invalid datetime format', 'ctx': None}]}}
```
If I force the installation of `starlette` 0.22.0 and rerun the test, everything works. This is a known bug of `starlette` 0.21.0 and it has been fixed in 0.22.0: 
- https://github.com/encode/starlette/pull/1953/files
- https://github.com/encode/starlette/releases/tag/0.22.0

## Solution

Bump `starlette` version in FastAPI to 0.22.0

## Note

I can make this small PR with the update as soon as you confirm/acknowledge this issue.

### Operating System

macOS

### FastAPI Version

0.87.0

### Python Version

3.10.0

**Comment by TheJumpyWizard** [comment]
There's a PR up for this now actually

**Comment by slafs** [comment]
The PR is https://github.com/tiangolo/fastapi/pull/5659
and this issue is then, most likely, a duplicate of https://github.com/tiangolo/fastapi/issues/5646.

**Comment by andreagrandi** [comment, marked as answer]
Thanks everyone for the feedback. It seems a duplicate of #5646 yes and since there is already a PR to fix this, I closed the issue already. Looking forward to 0.88.0 release 🫶
